serviceregistration/kubernetes/service_registration.go

```diff
@@ -0,0 +1,195 @@
+package kubernetes
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	sr "github.com/hashicorp/vault/serviceregistration"
+	"github.com/hashicorp/vault/serviceregistration/kubernetes/client"
+)
+
+const (
+	// Labels are placed in a pod's metadata.
+	labelVaultVersion = "vault-version"
+	labelActive       = "vault-active"
+	labelSealed       = "vault-sealed"
+	labelPerfStandby  = "vault-perf-standby"
+	labelInitialized  = "vault-initialized"
+
+	// This is the path to where these labels are applied.
+	pathToLabels = "/metadata/labels/"
+)
+
+func NewServiceRegistration(config map[string]string, logger hclog.Logger, state sr.State, _ string) (sr.ServiceRegistration, error) {
+	namespace, err := getRequiredField(logger, config, client.EnvVarKubernetesNamespace, "namespace")
+	if err != nil {
+		return nil, err
+	}
+	podName, err := getRequiredField(logger, config, client.EnvVarKubernetesPodName, "pod_name")
+	if err != nil {
+		return nil, err
+	}
+	return &serviceRegistration{
+		logger:       logger,
+		namespace:    namespace,
+		podName:      podName,
+		initialState: state,
+		retryHandler: &retryHandler{
+			logger:         logger,
+			namespace:      namespace,
+			podName:        podName,
+			patchesToRetry: make([]*client.Patch, 0),
```

Comment:
```suggestion
			patchesToRetry: nil,
```

Reply:
In line with that thinking, I did go through and omit it entirely everywhere else that's similar. https://github.com/hashicorp/vault/pull/8249/commits/84cd8ece1a2083c3f214cb3e9f49efa20e7432bd